### PR TITLE
Adding support for Taco code Coverage 

### PR DESCRIPTION
**Pull request by nisheetjain**

### CODE REVIEW

File: src/remotebuild/package.json [hunk 1]
```diff
@@ -42,12 +42,14 @@
...
         "mocha": "2.0.1",
+        "istanbul": "^0.4.0",
         "mkdirp": "0.3.5",
         "should": "4.3.0",
         "request": "2.36.0"
     },
     "scripts": {
-        "test": "mocha"
+        "test": "mocha",
+        "coverage": "istanbul cover --report json node_modules/mocha/bin/_mocha -- --reporter spec"
```
Comment (MSLaguana) [inline src/remotebuild/package.json]: Is there a reason you've used node_modules/mocha/bin/_mocha rather than node_modules/.bin/mocha ?

Comment (nisheetjain) [inline src/remotebuild/package.json]: yup on windows mocha ends up matching with mocha.cmd and istanbul fails to work with this. after lot of web search, found this is the right way to do this across platforms

Comment (MSLaguana) [inline src/remotebuild/package.json]: Ah ok, makes sense.

File: tools/GulpCoverageUtils.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        // read all the directories and prepare a flat list of coverageXXX.json we need to remap
+        return Q.all<string[]>(coverageResultsPaths.map(coverageResultsPath => {
+            coverageResultsPath = path.resolve(coverageResultsPath);
+            return Q.denodeify(fs.readdir)(coverageResultsPath)
+                .then(function(files: string[]): string[] {
+                    return files.filter(file => {
+                        return file.toLowerCase().indexOf("coverage-") === 0 && path.extname(file) === ".json";
+                    }).map(file => path.join(coverageResultsPath, file));
+                });
+        })).spread(function(...args: string[][]): string[] {
```
Comment (MSLaguana) [inline tools/GulpCoverageUtils.ts]: Why spread and then use ...args? You can just use a .then taking the `string[][]` as its first argument

File: tools/GulpCoverageUtils.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                return coverageFilesList.concat(coverageFiles);
+            });
+
+        });
+    }
+
+    private static normalizeCoveragePaths(coverageJsonsList: string[], coverageOutputDir: string, modulesRoot: string): Q.Promise<string[]> {
+        var index: number = 0;
+        // put all extra coverage files in a random location under coverage
+        var randomDir: string = Math.floor(Math.random() * 10000) + 1000 + "";
```
Comment (MSLaguana) [inline tools/GulpCoverageUtils.ts]: What's this for? we are hiding some files in a random place?

File: tools/GulpCoverageUtils.ts (hunk 2 above)
Comment (nisheetjain) [inline tools/GulpCoverageUtils.ts]: done

File: tools/GulpCoverageUtils.ts (hunk 3 above)
Comment (nisheetjain) [inline tools/GulpCoverageUtils.ts]: removed the random dir and replaced it with "secondary". something we delete post coverage

**Comment by meenakb** [comment]
:+1:

**Pull request merged**